metrics/metrics-core/src/main/java/org/apache/servicecomb/metrics/core/MonitorManager.java

```diff
@@ -0,0 +1,181 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.servicecomb.metrics.core;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.concurrent.Callable;
+
+import org.apache.servicecomb.foundation.common.concurrent.ConcurrentHashMapEx;
+import org.apache.servicecomb.foundation.metrics.MetricsConst;
+
+import com.netflix.config.DynamicPropertyFactory;
+import com.netflix.servo.BasicMonitorRegistry;
+import com.netflix.servo.MonitorRegistry;
+import com.netflix.servo.monitor.BasicCounter;
+import com.netflix.servo.monitor.BasicGauge;
+import com.netflix.servo.monitor.BasicTimer;
+import com.netflix.servo.monitor.Counter;
+import com.netflix.servo.monitor.Gauge;
+import com.netflix.servo.monitor.MaxGauge;
+import com.netflix.servo.monitor.Monitor;
+import com.netflix.servo.monitor.MonitorConfig;
+import com.netflix.servo.monitor.MonitorConfig.Builder;
+import com.netflix.servo.monitor.StepCounter;
+import com.netflix.servo.monitor.Timer;
+import com.netflix.servo.tag.Tag;
+import com.netflix.servo.tag.TagList;
+
+public class MonitorManager {
+
+  private final Map<String, Counter> counters;
+
+  private final Map<String, MaxGauge> maxGauges;
+
+  private final Map<String, Gauge> gauges;
+
+  private final Map<String, Timer> timers;
+
+  private final MonitorRegistry basicMonitorRegistry;
+
+  private static final MonitorManager INSTANCE = new MonitorManager();
+
+  public static MonitorManager getInstance() {
+    return INSTANCE;
+  }
+
+  private MonitorManager() {
+    this.counters = new ConcurrentHashMapEx<>();
+    this.maxGauges = new ConcurrentHashMapEx<>();
+    this.gauges = new ConcurrentHashMapEx<>();
+    this.timers = new ConcurrentHashMapEx<>();
+    this.basicMonitorRegistry = new BasicMonitorRegistry();
+    setupWindowTime();
+    registerSystemMetrics();
+  }
+
+  private void setupWindowTime() {
+    int time = DynamicPropertyFactory.getInstance().getIntProperty(MetricsConfig.METRICS_WINDOW_TIME, 5000).get();
+    System.getProperties().setProperty("servo.pollers", time > 0 ? String.valueOf(time) : "5000");
+  }
+
+  public Counter getCounter(boolean isStepCounter, String name, String... tags) {
```

Comment:
This method exposes the counter detail which is not right way.

Reply:
separate into two method named getCounter and getStepCounter